woocommerce/woocommerce
# Add `recordEvent` validation to Tracks package

### All Submissions:






### Changes proposed in this Pull Request:


This PR adds a validation to the [method recordEvent](https://github.com/woocommerce/woocommerce/blob/trunk/packages/js/tracks/src/index.ts#L27-L46). To have more context, please take a look at this P2 pcQJnU-z7-p2

### How to test the changes in this Pull Request:

1. Checkout this branch and smoke test the React pages (OBW, Homescreen, etc).
2. Verify that nothing is broken.
3. I created the branch `add/validation_for_event_tracking_test` to test this PR more easily.
4. Go to the OBW.
5. After checking out the branch, the event `badEventName` should fail its validation. (Be sure that you have the `NODE_ENV` set as `development`)
6. Verify that the error message `A valid event name must be specified. The event name: "badEventName" is not valid.` is visible in the devtools console.
7. Run tests and verify they finish correctly.

### Other information:




### FOR PR REVIEWER ONLY:
